### ISSUE TITLE
Bug: dont make request to /api/recording-rules/writer when viewing a pubdash

### ISSUE THREAD

**Issue by owensmallwood** [grafana/grafana#62357]
When viewing a pubdash while unauthed, we still make a request to `/api/recording-rules/writer` which requires a signed in user.

If we're viewing a pubdash, don't make the request.

**Comment by Aksh-Bansal-dev** [comment]
I want to work on this issue but I can't seem to find where we are requesting `/api/recording-rules/writer` in the codebase.

**Comment by ashutosh887** [comment]
I would like to work on this Issue! @owensmallwood !

**Comment by MTLChrisLEE** [comment]
Hi @owensmallwood 

I tried to reproduce the issue in my local env but I was unable to do so.
Here are the steps I did.
 1. Create custom.ini in conf folder
 2. [feature_toggles]
     publicDashboards = true
 3.  Log in as admin and create a dashboard
 4. Change the dashboard's visibility to Public
![image](https://user-images.githubusercontent.com/25248744/218245070-2e271815-82f1-440a-b34f-2f7d4df5f8e8.png)
  5. Visit the URL incognito mode.
  
  What am I doing wrong?
  
  Thanks
